src/Pact/Types/PactValue.hs

```diff
@@ -90,3 +106,211 @@ toPactValueLenient t = case toPactValue t of
   Right (PLiteral (LInteger l)) -> PLiteral (LDecimal (fromIntegral l))
   Right v -> v
   Left _ -> PLiteral $ LString $ renderCompactText t
+
+
+
+-- |  Estimate of number of bytes needed to represent data type
+--
+-- Assumptions: GHC, 64-bit machine
+-- General approach:
+--   Memory Consumption = Constructor Header Size + Cost of Constructor Field(s)
+--   Cost of Constructor Field(s)* = 1 word per field + cost of each field's value
+-- (*) See Resource 2 for exceptions to these rules
+-- Resources:
+-- 1. http://wiki.haskell.org/GHC/Memory_Footprint
+-- 2. https://stackoverflow.com/questions/3254758/memory-footprint-of-haskell-data-types
+
+class SizeOf t where
+  sizeOf :: t -> Bytes
+
+
+type Bytes = Int64
+
+-- TOOD Figure out how to detect machine arch
+-- | "word" is 4 bytes on 32-bit arch, but 8 bytes on 64-bit
+wordSize32, wordSize64, wordSize :: Bytes
+wordSize32 = 4
+wordSize64 = 8
+wordSize = wordSize64
+
+-- | Constructor header is 1 word
+headerCost :: Bytes
+headerCost = 1 * wordSize
+
+-- | In general, each constructor field costs 1 word
+constructorFieldCost :: Int64 -> Bytes
+constructorFieldCost numFields = numFields * wordSize
+
+-- | Total cost for constructor
+constructorCost :: Int64 -> Bytes
+constructorCost numFields = headerCost + (constructorFieldCost numFields)
+
+
+instance SizeOf PactValue where
+  sizeOf (PLiteral l) = (constructorCost 1) + (sizeOf l)
+  sizeOf (PList v) = (constructorCost 1) + (sizeOf v)
+  sizeOf (PObject o) = (constructorCost 1) + (sizeOf o)
+  sizeOf (PGuard g) = (constructorCost 1) + (sizeOf g)
+
+instance SizeOf Literal where
+  sizeOf (LString t) = (constructorCost 1) + (sizeOf t)
```

Comment:
Text has several other fixed costs

  - pointer to Array constructor, pointer to bytearray#
  - bytearray# has an internal word-sized length
  - text has an offset and length for slicing

If you add it all up I think you need to account for 5 extra words

Reply:
`sizeOf t` here will trigger this instance of [`SizeOf Text` ](https://github.com/kadena-io/pact/pull/676/files#diff-e7679ea20830b8fa20610f2fdf9ac01bR286)

I used the formula for the Text memory footprint specified here: http://wiki.haskell.org/GHC/Memory_Footprint aka
`6 words + 2N bytes`

Would there be 5 extra words on top of the above?

Reply:
No I think you are ok